File: backend/satisfactory-blueprint/satisfactory-blueprint/src/main/java/com/satisfactory/blueprint/dto/CreateTransportRouteRequest.java
package com.satisfactory.blueprint.dto;

import com.satisfactory.blueprint.entity.enums.VehicleType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateTransportRouteRequest {
    private Long planId;
    private String stationName;
    private VehicleType vehicleType;
    private Double capacity;
    private Long assignedItemId;
    private List<Double> cartCapacities;
}

File: frontend/satisfactory-blueprint-designer/src/types/createTransportRouteRequest.ts
// src/types/createTransportRouteRequest.ts
/**
 * Mirrors com.satisfactory.blueprint.dto.CreateTransportRouteRequest
 */
export interface CreateTransportRouteRequest {
  planId: number;
  stationName: string;
  vehicleType: string;
  capacity: number;
  assignedItemId: number;
  cartCapacities: number[];
}
